core/src/main/java/io/grpc/internal/Stream.java

```diff
@@ -77,4 +79,10 @@
    * result in excessive buffering within the transport.
    */
   boolean isReady();
+
+  /**
+   * Sets the default message encoder for messages on this stream.
+   * @param c the compressor / decompressor
+   */
+  void setDefaultCodec(Codec c);
```

Comment:
But what is "default" about it. Why is it not `setCodec`? With "default," I expect it to be overridden with another `Codec` at some other point.